### PR TITLE
feat: dart language support

### LINKED ISSUE #45: new language: Dart
Add Dart as a supported language to detect.

You can learn Dart here:
- https://learnxinyminutes.com/docs/dart/
- https://www.youtube.com/watch?v=5xlVP04905w
- https://dart.dev/tutorials

Some test cases:
- https://rosettacode.org/wiki/Hello_world/Text#Dart
- https://rosettacode.org/wiki/Happy_numbers#Dart
- https://rosettacode.org/wiki/FizzBuzz#Dart
- https://rosettacode.org/wiki/Guess_the_number#Dart
- https://rosettacode.org/wiki/Palindrome_detection#Dart
- https://rosettacode.org/wiki/Sorting_algorithms/Bubble_sort#Dart
- https://rosettacode.org/wiki/Sorting_algorithms/Merge_sort#Dart
- https://rosettacode.org/wiki/Sorting_algorithms/Heapsort#Dart
- https://rosettacode.org/wiki/Sorting_algorithms/Quicksort#Dart
- https://rosettacode.org/wiki/Regular_expressions#Dart
- https://rosettacode.org/wiki/Pascal%27s_triangle#Dart
- https://rosettacode.org/wiki/Perfect_numbers#Dart
- https://rosettacode.org/wiki/N-queens_problem#Dart

### PR DESCRIPTION
**Pull request by aldy505**
This is pretty painful

Closes #45

### CODE REVIEW

File: src/languages/cs.ts [hunk 1]
```diff
@@ -39,4 +39,6 @@ export const CS: LanguagePattern[] = [
   { pattern: /System\.(in|out)\.\w+/, type: 'not' },
   // Avoiding Ruby confusion
   { pattern: /\bmodule\s\S/, type: 'not' },
+  // Avoiding Dart confusion
+  { pattern: /^import\s("|')dart:\w+("|')/, type: 'not' },
```
Comment (farhan443) [inline src/languages/cs.ts:43]: import statement can't be preceded with a whitespace?

File: src/languages/dart.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    type: 'keyword.control',
+  },
+  { pattern: /\bget\s\w+=>\w+/, type: 'keyword.control' },
+  { pattern: /^\s*@override$/, type: 'keyword.control' },
+  { pattern: /\bset\s\w+\(.+\)/, type: 'keyword.control' },
+  { pattern: /^\s*Future<w+>\s\w+\(.+\)\sasync/, type: 'keyword.control' },
+  { pattern: /^\s*await\sfor/, type: 'keyword.control' },
+  { pattern: /^\s*typedef\s.+\s=/, type: 'keyword.control' },
+  // Avoiding confusion with C
+  { pattern: /\blong\b/, type: 'not' },
```
Comment (farhan443) [inline src/languages/dart.ts:40]: Can the `long` keyword surrounded by punctuation? If not, you might want to use `\s` instead.

File: src/languages/c.ts [hunk 3]
```diff
@@ -48,4 +48,6 @@ export const C: LanguagePattern[] = [
   },
   // Avoiding Lua confusion
   { pattern: /local\s(function|\w+)?/, type: 'not' },
```
Comment (farhan443) [inline src/languages/c.ts:50]: Shouldn't it be `\s+` instead of `\s`?

File: src/languages/cs.ts (hunk 1 above)
Comment (aldy505) [reply src/languages/cs.ts:43]: bisa aja sih sebenernya, but don't know

**Comment by aldy505** [comment]
woi gimana @farhan443

**Pull request merged**